### PR TITLE
Breakdown score

### PR DESCRIPTION
**Pull request by SunandanBose**
Problem solved : https://github.com/kumarRaj/stocks/issues/34

### CODE REVIEW

File: backend/src/main/java/com/project/stocks/service/ScoreBuilder.java [hunk 1]
```diff
@@ -18,16 +18,19 @@ public class ScoreBuilder {
 
 
     public ScoreBuilder withPE(Integer pe) {
+        int sum = 0;
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/service/ScoreBuilder.java:21]: What does this sum up to? It's just the PE value

File: backend/src/main/java/com/project/stocks/service/ScoreBuilder.java [hunk 2]
```diff
@@ -45,27 +48,37 @@ public Score build() {
...
+        score.addValue(value);
+        score.getScoreBreakdown().setRevenue(value);
         return this;
     }
 
     public ScoreBuilder withBorrowings(List<YearInfo> borrowingsDetails) {
-        score.addValue(calculateYearlyStatistics(borrowingsDetails, Decreasing));
+        int value = calculateYearlyStatistics(borrowingsDetails, Decreasing);
+        score.addValue(value);
+        score.getScoreBreakdown().setRevenue(value);
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/service/ScoreBuilder.java:74]: This should be borrowings right?

File: backend/src/main/java/com/project/stocks/service/ScoreBuilder.java [hunk 3]
```diff
@@ -45,27 +48,37 @@ public Score build() {
 
 
     public ScoreBuilder withOPM(List<YearInfo> opmList) {
-        score.addValue(calculateYearlyStatistics(opmList, Logic.Increasing));
+        int value = calculateYearlyStatistics(opmList, Logic.Increasing);
+        score.addValue(value);
+        score.getScoreBreakdown().setOperatingProfitMargin(value);
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/service/ScoreBuilder.java:53]: How I would write this code would be to have a score breakdown property in the builder,.
Use `scoreBreakdown.setOperatingProfitMargin(value);` 

Then in the build method just add
`score.setScoreBreakdown(scoreBreakdown);`

**Comment by kumarRaj** [comment]
Do we still need the vercel deployment?

File: backend/src/main/java/com/project/stocks/service/ScoreBuilder.java [hunk 4]
```diff
@@ -18,16 +18,19 @@ public class ScoreBuilder {
...
-            score.addValue(3);
+            sum = 3;
         else if (pe >= 61 && pe <= 80)
-            score.addValue(2);
+            sum = 2;
         else
-            score.addValue(1);
+            sum = 1;
+        score.addValue(sum);
+        score.getScoreBreakdown().setPe(sum);
```
Comment (kumarRaj) [inline backend/src/main/java/com/project/stocks/service/ScoreBuilder.java:33]: Can we also have tests for score breakdown calculation?

**Review by kumarRaj — CHANGES_REQUESTED** [review]
Have left comments

**Comment by SunandanBose** [comment]
yes, I still need vercel deployment. I will fix the UI soon.

**Review by kumarRaj — APPROVED** [review]
Looks good

**Pull request merged**